cmd/commands/server/command_server.go

```diff
@@ -69,10 +68,14 @@ func (cmd *Command) Start() (err error) {
 		return err
 	}
 
-	serviceLocation, err := detectCountry(cmd.ipResolver, cmd.locationDetector)
+	locationDetector := location.NewDetectorWithLocationResolver(cmd.ipResolver, cmd.locationResolver)
```

Comment:
`locationDetector` should be constructed and injected already

Reply:
done